driver-core/src/main/com/mongodb/internal/connection/ConcurrentPool.java

```diff
@@ -167,25 +170,49 @@ public void prune() {
         }
     }
 
-    public void ensureMinSize(final int minSize, final boolean initialize) {
+    /**
+     * Try to populate this pool with items so that {@link #getCount()} is not smaller than {@code minSize}.
+     * The {@code postCreate} action throwing a exception causes this method to stop and re-throw that exception.
+     *
+     * @param postCreate A transforming action applied to non-{@code null} new items.
+     *                   This action may return an item as is, may modify the item,
+     *                   or throw an {@link Exception}. In the latter case the action must release
+     *                   resources associated with the item.
+     */
+    public void ensureMinSize(final int minSize, @Nullable final UnaryOperator<T> postCreate) {
         while (getCount() < minSize) {
-            if (!acquirePermit(10, TimeUnit.MILLISECONDS)) {
+            if (!acquirePermit(0, TimeUnit.MILLISECONDS)) {
```

Comment:
The change is intentional. 0 means "try and return without waiting if a permit is not available". These permits are used to enforce `maxSize`. Not observing an available permit means that `maxSize` either has been reached or is about to be reached (a permit is acquired before creating a new element and adding it to the pool). `ensureMinSize` should return in such a situation instead of waiting. I do not have an explanation of why the timeout was not 0 previously.

Reply:
Ack